### DISCUSSION TITLE
Multiple station lists into METplus job

### DISCUSSION THREAD

**Discussion by robdarvell** [dtcenter/METplus#1208] [Configuration] [answered]
Hi,
It is possible to include multiple area mask fields into your GridStat job, but is the same possible for station lists and their use within PointStat ?  (not saying that my syntax of following the same idea as for the area masks was wrong).

For example, we would like to say generate surface temperature results for a list of defined stations say over the UK and a list of defined stations over the US. And run these within the same job. This is slightly different from storing the raw paired data and then using another tool to calculate the stats from that info.

Is that something that can be done already, might be useful in the future ?

Rob

**Comment by CPKalb** [comment]
Hi Rob:

I wanted to clarify to be sure I understand.  Do you want separate statistics over the UK and US?  If so, point_stat does accept masking regions.  So, you can give a masking region (or something like a square region defined by latitude/longitude) for each of these areas.  Would something like this work for your data?

Christina

**Comment by robdarvell** [comment]
Hi Christina,

Not quite. The masking region would include all stations found within that area mask (unless say they fail QC, elevation stuff etc) as I understand. What I am looking for is a defined list of stations, which is just a subset of those found within the area mask. So for example, for WMO CBS we have station lists for Radiosonde reports across Europe, N.America for example which contain just those locations which have been accepted for inclusion in the stats and we generate those stats at the same time as different codes. By being able to do this we are only having to read the forecast data in once. I can see how you can run PointStat several times with a different set of stations each time, but was trying to avoid that.

Rob

**Reply by CPKalb** [reply]
Hi Rob:

One thing you could do is change the message type for the stations you want to retain.  I don't know if this would be easier/faster than just running point_stat several times though.  I am double checking whether point_stat will take a station list.

Christina

**Comment by CPKalb** [comment]
Hi Rob:

I apologize for giving you incorrect information.  You can give a list of stations by specifying sid in the mask dictionary in the parameter file.  It would look something like this:
mask = {
   grid  = [ ];
   poly  = [  ];
   sid     = [ "CONUS.stations","NH.stations","Fred.stations" ];         where CONUS.stations, NH.stations, Fred.stations are each a station list
}

The sid can take an ascii file with the list of stations (as shown above), or a list of stations, which would look something like this:
mask = {
   grid  = [ ];
   poly  = [  ];
   sid     = [ "KDEN", "KDET", "SWS01" ];
}

More information can be found here https://met.readthedocs.io/en/develop/Users_Guide/config_options.html#mask

**Reply by TaraJensen** [reply]
@georgemccabe  - Rob is actually trying to do this with METplus wrappers.  Can you jump in and confirm the syntax for multiple station lists using the wrappers?

**Comment by robdarvell** [comment]
OK, have got a little further:

If I include all of the station lists on a single line in my METplus wrapper it all works: (note this will probably wrap)
POINT_STAT_STATION_ID = /data/users/cfrd/METplus_playaround/StationLists/2011.stns,                    /data/users/cfrd/METplus_playaround/StationLists/2012.stns

However, if I do it on multiple lines (as I do for creating multiple area mask lists) then it fails with a whitespace issue:
POINT_STAT_STATION_ID = /data/users/cfrd/METplus_playaround/StationLists/2011.stns,\
                        /data/users/cfrd/METplus_playaround/StationLists/2012.stns
(same outcome if I start the second line from the left-hand edge rather than how it looks here)

10/20 16:45:31.402 metplus INFO: POINT_STAT_POLY=[]
10/20 16:45:31.402 metplus INFO: POINT_STAT_STATION_ID=["/data/users/cfrd/METplus_playaround/StationLists/2011.stns", "
/data/users/cfrd/METplus_playaround/StationLists/2012.stns"]
10/20 16:45:31.402 metplus INFO: REGRID_TO_GRID=NONE

...

DEBUG 3: /data/users/cfrd/METplus_playaround/StationLists/2012.stns
WARNING: 
WARNING: parse_sid_mask() -> unable to process "
WARNING: /data/users/cfrd/METplus_playaround/StationLists/2012.stns" as a file name and processing it as a single station ID mask instead.
WARNING: 
DEBUG 4: parse_sid_mask() -> storing single station ID mask "
DEBUG 4: /data/users/cfrd/METplus_playaround/StationLists/2012.stns"
ERROR  : 
ERROR  : parse_sid_mask() -> masking station ID string can't contain whitespace or slashes "
ERROR  : /data/users/cfrd/METplus_playaround/StationLists/2012.stns".

**Comment by georgemccabe** [comment, marked as answer]
Hi @robdarvell, you do not need to provide the square brackets around the value for the mask.sid list. Removing the []s should fix this issue. I have thought about enhancing the logic to properly handle when the braces are included since this has happened to other users. I created issue #1212 to make this fix.

On a side note, POINT_STAT_STATION_ID is the old name for the variable. It will still work fine if set, but it is preferred to use the new name of POINT_STAT_MASK_SID that more closely aligns to the mask.sid variable in the PointStat config file.

Please let me know if you have any other issues.

**Comment by robdarvell** [comment]
Thanks all. Have now got something running which will do 2 area mask lists and 2 station mask lists in the same job (as a test).

I think the main confusion I had was around setting up some of the templates. For example: what I was trying to do was (in my METplus wrapper)...

POINT_STAT_MASK_POLY=/path/to/mask_a,(backslash)
                                           /path/to/mask_b

POINT_STAT_MASK_SID=/path/to/station_a,(backslash)
                                        /path/to/station_b

This was primarily to make the wrapper script look a bit neater. Problem I was running into was the ,\ at the end of the row only works with the mask poly list, you don't need the \ on the sid list. Was wondering whether a more consistent approach may assist ?

**Reply by georgemccabe** [reply]
The METplus config file parser is built upon the Python ConfigParser package. According to the rules listed in their documentation, multiple line values can be set by indenting the lines after the first using spaces or tabs (See https://docs.python.org/3/library/configparser.html#supported-ini-file-structure). I don't think the backslash is needed. Here is an example in a METplus config file:

https://github.com/dtcenter/METplus/blob/b2e65bc3a6251d8c4b4e88d3e00a9eb84827a050/parm/use_cases/met_tool_wrapper/EnsembleStat/EnsembleStat.conf#L254-L259

**Comment by JohnHalleyGotway** [comment]
@robdarvell I'd like to mark this discussion as being answered, and then lock it to prevent future posts.
But I wanted to give you a chance to take a look and confirm that your initial question has been answered.
Please let me know.

**Reply by robdarvell** [reply]
Yes, I have managed to do what I require to do with this question. It was more a case of me getting confused with continuation lines within the METplus wrapper side of things which is where my confusion mainly came from.
